Components/CategoryResolver/AutoCategoryResolver.php

```diff
@@ -155,6 +166,22 @@ public function storeRemoteCategories(array $categories, $articleId)
     }
 
     /**
+     * @param $leaf
+     */
+    public function addLeaf(Category $leaf)
+    {
+        $this->leafCollection[] = $leaf;
+    }
+
+    /**
+     * @return Category[]
+     */
+    public function getLeafCollection()
+    {
+        return $this->leafCollection;
```

Comment:
Please refactor this method, ```public function getLeafCollection(array $tree)```
store leaf categories in array and return them.
You don't need to use property for that.